examples/thread_duel/main.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2020 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <thread.h>
+#include <sched.h>
+
+#include <math.h>
+
+#if MODULE_ZTIMER_USEC
+    #include "ztimer/xtimer_compat.h"
+#else
+    #include "xtimer.h"
+#endif
+
+#include "sched_round_robin.h"
+
+#define PRINT_STEPS 10
+#define WORK_SCALE  1000
+
+static void bad_wait(uint32_t us)
+{
+    uint32_t start = xtimer_now_usec();
+    /*keep the CPU busy waiting for some time to pass simulate working*/
+    while ((xtimer_now_usec() - start) < us){
+        /*do nothing*/
+    }
```

Comment:
Isn't this the same as `xtimer_spin`?

Reply:
yes but xtimer spin is internal to xtimer (at least it was when i created this)